src/main/java/org/folio/rest/impl/TenantReferenceAPI.java

```diff
@@ -0,0 +1,249 @@
+package org.folio.rest.impl;
+
+import static java.util.stream.Collectors.toList;
+import static org.folio.rest.RestVerticle.MODULE_SPECIFIC_ARGS;
+
+import io.vertx.core.*;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import java.io.*;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import javax.ws.rs.core.Response;
+import org.apache.commons.io.IOUtils;
+import org.folio.rest.jaxrs.model.Parameter;
+import org.folio.rest.jaxrs.model.TenantAttributes;
+
+public class TenantReferenceAPI extends TenantAPI {
+  private static final Logger log = LoggerFactory.getLogger(TenantReferenceAPI.class);
+
+  private static final String JAR_PROTOCOL = "jar";
+  private static final String FILE_PROTOCOL = "file";
+  private static final String RESOURCES_PATH = "data/";
+  private static final String ORDERS_STORAGE_PREFIX_URL = "/orders-storage/";
+  private static final String PARAMETER_LOAD_SAMPLE = "loadSample";
+  private HttpClient httpClient;
+
+
+  @Override
+  public void postTenant(TenantAttributes tenantAttributes, Map<String, String> headers, Handler<AsyncResult<Response>> hndlr, Context cntxt) {
+    log.info("postTenant");
+
+    httpClient = cntxt.owner().createHttpClient();
+    super.postTenant(tenantAttributes, headers, res -> {
+      //if a system parameter is passed from command line, ex: loadSample=true that value is considered,
+      //Priority of Parameter Tenant Attributes > command line parameter > default (false)
+      boolean loadSample = Boolean.parseBoolean(MODULE_SPECIFIC_ARGS.getOrDefault(PARAMETER_LOAD_SAMPLE,
```

Comment:
There's no need to read/parse the command line arg each time... just do it once

Reply:
right, it's done every time the endpoint is called.  Since this is a command line argument it's not going to change unless the application is restarted, so you can read this in as a static variable...

Reply:
Ahh right, Done